File: Source/ses_io/src/wrappers/fortran/wrap_ses_get_materials.f90


#undef DEBUG_WRAP

function ses_get_materials_f90(the_handle, pt_size)

   implicit none

   SES_FILE_HANDLE, intent(IN) :: the_handle
   LONG, dimension(1), target :: the_size
   
   SES_MATERIAL_ID_REFERENCE :: ses_get_materials_f90
   SES_MATERIAL_ID, dimension(1000), target :: actual_materials
#ifdef DEBUG_32BIT
   integer(kind = 4), dimension(1000), target :: actual_table_4
   integer(kind = 8), dimension(1000), target :: actual_table_8
   integer(kind = 4), dimension(:), pointer :: return_4
   integer(kind = 8), dimension(:), pointer :: return_8
#endif

   integer :: ses_get_materials, dummy
   LONG_ARRAY_POINTER :: pt_size   !  has no memory here

   ses_get_materials_f90=>actual_materials
#ifdef DEBUG_32BIT
   return_4=>actual_table_4
   return_8=>actual_table_8
#endif

   
#ifdef DEBUG_WRAP
   print *, "wrap_ses_get_materials_f90:  the_handle is ", the_handle
#endif

#ifdef DEBUG_32BIT
   dummy = ses_get_materials(the_handle, pt_size, return_4)
   ses_get_materials_f90 = return_4
#else
   dummy = ses_get_materials(the_handle, pt_size, ses_get_materials_f90)
#endif
#ifdef DEBUG_WRAP   
   print *, "wrap_ses_get_materials_f90:  dummy is ", dummy
   print *, "wrap_ses_get_materials_f90:  ses_get_materials_f90(1) is ", ses_get_materials_f90(1)
   print *, "wrap_ses_get_materials_f90:  the_size is ", pt_size(1)
#endif

   
  
end function ses_get_materials_f90

File: Source/ses_io/src/wrappers/fortran/wrap_ses_get_materials.c


#include "ses_defines.h"
#include "ses_globals.h"
#include "ses_internals.h"
#include "ses_externs.h"

#include "stdio.h"
#include <stdlib.h>

#include <stdio.h>

#undef DEBUG_WRAP

#ifdef LC_UNDER
int ses_get_materials_(ses_file_handle* pt_handle, long* the_size, ses_material_id_reference the_return) {
#else

#ifdef UC_UNDER
int SES_GET_MATERIALS_(ses_file_handle* pt_handle, long* the_size, ses_material_id_reference the_return) {
#else

#ifdef LC_NOUNDER
int ses_get_materials(ses_file_handle* pt_handle, long* the_size, ses_material_id_reference the_return) {
#else

#ifdef UC_NOUNDER
int SES_GET_MATERIALS(ses_file_handle* pt_handle, long* the_size, ses_material_id_reference the_return) {
#endif

#endif
#endif
#endif

  int return_value = 0;

 
  ses_file_handle the_handle = *pt_handle;
#ifdef DEBUG_WRAP
  printf("wrap_ses_get_materials.c:  the_handle is %d\n", the_handle);
#endif

  ses_material_id_reference dummy_array;
  dummy_array = ses_get_materials(the_handle, the_size);

#ifdef DEBUG_WRAP
  printf("wrap_ses_get_materials.c:  the returned size is %d\n", *the_size);
#endif

  int i = 0;
  
  for(i=0; i < *the_size; i++) {
    the_return[i] = dummy_array[i];
  }
#ifdef DEBUG_WRAP 
  printf("wrap_ses_get_materials.c:  the_return[0] is %d\n", the_return[0]);
  printf("wrap_ses_get_materials.c:  the_size[0] is %d\n", the_size[0]); 
#endif
  return_value = *the_size;
  return return_value;
}

 
